src/core/thread/mle_router.cpp

```diff
@@ -599,6 +599,15 @@ Error MleRouter::SendLinkRequest(Neighbor *aNeighbor)
     return error;
 }
 
+void MleRouter::InitNeighbor(Neighbor &aNeighbor, const RxInfo &aRxInfo)
```

Comment:
Should we try to use this here as well?
https://github.com/openthread/openthread/blob/6e975c587556ba49964153a30436978da23b0aff/src/core/thread/mle.cpp#L3276

Reply:
Good idea. It may increase MTD build by a bit (since we will add a new method and only use it once under MTD) but I think it is ok and worth the simplification in FTD build.

Updated in new push.